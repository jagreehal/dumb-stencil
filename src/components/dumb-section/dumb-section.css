:host {
  display: flex;
}

.c-section {
  padding: 2rem;
}

.c-section__header {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-bottom: 2rem;
}

.c-section__title {
  font-size: 2rem;
  margin: 0;
}

.c-section__description {
  margin: 0;
}
