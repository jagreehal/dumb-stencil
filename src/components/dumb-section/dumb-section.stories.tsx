import '@stencil/core';

import { storiesOf } from '@storybook/html';

const stories = storiesOf('Containers/Section', module);

stories.add('Default', () => {
  return `<dumb-section
  section-title="Section Title"
  section-description="This is a description of what the section is">
    This is the section body, where you can put any content or include other components.
  </dumb-section>`;
});
