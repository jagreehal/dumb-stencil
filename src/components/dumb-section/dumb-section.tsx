import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'dumb-section',
  styleUrl: 'dumb-section.css',
  shadow: true
})
export class DumbSection {
  @Prop() sectionTitle: string;
  @Prop() sectionDescription: string;

  render() {
    return (
      <section class="c-section">
        <header class="c-section__header">
          <h2 class="c-section__title">{this.sectionTitle}</h2>
          <p class="c-section__description">{this.sectionDescription}</p>
        </header>
        <div class="c-section__body">
          <slot />
        </div>
      </section>
    );
  }
}
