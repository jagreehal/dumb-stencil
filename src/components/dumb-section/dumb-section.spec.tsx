import { TestWindow } from '@stencil/core/testing';
import { DumbSection } from './dumb-section';

describe('dumb-section', () => {
  let element;
  let window;

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbSection],
      html: `
      <dumb-section
        section-title="Section Title"
        section-description="This is a description of what the section is">
          This is the section body, where you can put any content or include other components.
      </dumb-section>`
    });
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
