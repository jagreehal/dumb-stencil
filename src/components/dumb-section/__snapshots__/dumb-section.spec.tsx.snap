// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-section should match snapshot 1`] = `
<dumb-section
  class="hydrated"
  section-description="This is a description of what the section is"
  section-title="Section Title"
>
  
  
  <section
    class="c-section"
  >
    <header
      class="c-section__header"
    >
      <h2
        class="c-section__title"
      >
        Section Title
      </h2>
      <p
        class="c-section__description"
      >
        This is a description of what the section is
      </p>
    </header>
    <div
      class="c-section__body"
    >
      
      
          This is the section body, where you can put any content or include other components.
      
    </div>
  </section>
</dumb-section>
`;
